lib/rack/honeycomb/middleware.rb

```diff
@@ -43,6 +49,15 @@ def call(env)
         add_field(ev, 'HTTP_STATUS', status)
         add_field(ev, 'REQUEST_TIME_MS', (request_ended_at - request_started_at) * 1000)
 
+        # Pull arbitrary metadata off `env` if the caller attached anything
+        # inside the Rack handler.
+        env.each_pair do |k, v|
+          if k.start_with?(HEADER_PREFIX)
```

Comment:
Probably-silly question: does rack _guarantee_ that all env keys must be strings, or are dot-separated strings just the most idiomatic choice? I'm imagining that `env` is a plain Hash without any method override for `[]=` that would `.to_s` keys, and wondering if it's not safe to assume they would all respond to `start_with?`.

Reply:
Mmmmm I bet you're right. I'll add an `.is_a?(String)` check. 🙏